dotnetv3/Cognito/Actions/CognitoWrapper.cs

```diff
@@ -0,0 +1,362 @@
+﻿// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier:  Apache-2.0
+
+using System.Net;
+
+namespace CognitoActions;
+
+/// <summary>
+/// Methods to perform Amazon Cognito Identity Provider actions.
+/// </summary>
+public class CognitoWrapper
+{
+    private readonly IAmazonCognitoIdentityProvider _cognitoService;
+
+    /// <summary>
+    /// Constructor for the wrapper class containing Amazon Cognito actions.
+    /// </summary>
+    /// <param name="cognitoService"></param>
+    public CognitoWrapper(IAmazonCognitoIdentityProvider cognitoService)
+    {
+        _cognitoService = cognitoService;
+    }
+
+    // snippet-start:[Cognito.dotnetv3.ListUserPools]
+    /// <summary>
+    /// List the Amazon Cognito user pools for an account.
+    /// </summary>
+    /// <returns>A list of UserPoolDescriptionType objects.</returns>
+    public async Task<List<UserPoolDescriptionType>> ListUserPoolsAsync()
+    {
+        var response = new ListUserPoolsResponse();
+        var request = new ListUserPoolsRequest { MaxResults = 10 };
```

Comment:
This looks like it is paging using the NextToken, but it looks like there is a paginator for that in cognitoService.Paginators. Also, if the goal is to list all pages, do you want a MaxResults of 10?

Reply:
Replaced with a paginator.